model/source_http.go

```diff
@@ -14,7 +25,46 @@ type SourceCreateRequest struct {
 	AppCreationWorkflow string  `json:"app_creation_workflow"`
 	AvailabilityStatus  string  `json:"availability_status"`
 
-	SourceTypeID *int64 `json:"source_type_id"`
+	SourceTypeID    *int64      `json:"-"`
+	SourceTypeIDRaw interface{} `json:"source_type_id"`
+}
+
+// Validate validates that the required fields of the SourceCreateRequest request hold proper values. In the specific
+// case of the UUID, if an empty or nil one is provided, a new random UUID is generated and appended to the request.
+func (req *SourceCreateRequest) Validate() error {
+	if req.Name == nil || *req.Name == "" {
```

Comment:
@MikelAlejoBR why it requires both checks ? isn't enough just e.g. `req.Name == nil` ?

Reply:
The Rails app actually had an :allow_blank => false for the :name attribute, so that's why we're performing both checks: models/source.rb#L23